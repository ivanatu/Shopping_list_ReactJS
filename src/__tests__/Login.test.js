import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ReactDOM from 'react-dom' 
import toJson from 'enzyme-to-json'
import renderer from 'react-test-renderer'
import Login from '../components/login'
import moxios from 'moxios'
import sinon from 'sinon';
import {MemoryRouter} from 'react-router-dom'

describe('<Login/>', () => {
   
    const wrapper = shallow( <Login/> );

    it('renders without crashing', () => {
    shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
    });
    it('has a valid snapshot', () => {
        const component = renderer.create(
            <MemoryRouter><Login/></MemoryRouter>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
   
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="login"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="login-screen"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="app-title"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="login-form"/>))
    });
    it('should render <form> without throwing an error', () => {
        expect(wrapper.exists(<form className="register-form"/>))
    });
    it('renders three div jsx elements', () => {
        expect(wrapper.find("div")).toHaveLength(6);        
    });
    it('renders input tags  elements', () => {
        expect(wrapper.find("input")).toHaveLength(2);        
    });
    it('renders button elements', () => {
        expect(wrapper.find("button")).toHaveLength(1);        
    });
    it('renders link elements', () => {
        expect(wrapper.find("Link")).toHaveLength(1);        
    });
    it('should render label without throwing an error', () => {
        expect(wrapper.exists(<label className="login-field-icon fui-user"/>))
    });
    it('should render label without throwing an error', () => {
        expect(wrapper.exists(<label className="login-field-icon fui-lock"/>))
    });
    it('should render form inputs', () => {
        expect(wrapper.find('#email').length).toEqual(1);
        expect(wrapper.find('#password').length).toEqual(1);
    });
    it('input should respond to change event and change the state', () => {
        wrapper.find('#email').simulate('change', { target: { name: 'email', value: 'admin@gmail.com' } });
        expect(wrapper.state('email')).toEqual('admin@gmail.com')
    });
    it('input should respond to change event and change the state', () => {
        wrapper.find('#password').simulate('change', { target: { name: 'password', value: 'baron1234' } });
        expect(wrapper.state('password')).toEqual('baron1234')
    });
    it('renders login form and submits data', () =>{
        wrapper.find("#login_form").simulate('submit', {preventDefault(){}});
        expect(toJson(wrapper)).toMatchSnapshot();
    });
 
});