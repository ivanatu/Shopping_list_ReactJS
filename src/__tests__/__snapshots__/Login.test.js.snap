// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="login-screen"
      >
        <div
          className="app-title"
        >
          <h1>
            Login
          </h1>
        </div>
        <div
          className="login-form"
        >
          <form
            className="register-form"
            method="POST"
            name="login_form"
            onSubmit={[Function]}
          >
            <div
              className="control-group"
            >
              <input
                className="login-field"
                name="email"
                placeholder="email"
                required={true}
                type="text"
              />
              <label
                className="login-field-icon fui-user"
                htmlFor="login-name"
              />
            </div>
            <div
              className="control-group"
            >
              <input
                className="login-field"
                name="password"
                placeholder="password"
                required={true}
                type="password"
              />
              <label
                className="login-field-icon fui-lock"
                htmlFor="login-pass"
              />
            </div>
            <button
              className="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </form>
          <Link
            replace={false}
            to="/register"
          >
             Register new Account
          </Link>
        </div>
      </div>,
      "className": "login",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="app-title"
          >
            <h1>
              Login
            </h1>
          </div>,
          <div
            className="login-form"
          >
            <form
              className="register-form"
              method="POST"
              name="login_form"
              onSubmit={[Function]}
            >
              <div
                className="control-group"
              >
                <input
                  className="login-field"
                  name="email"
                  placeholder="email"
                  required={true}
                  type="text"
                />
                <label
                  className="login-field-icon fui-user"
                  htmlFor="login-name"
                />
              </div>
              <div
                className="control-group"
              >
                <input
                  className="login-field"
                  name="password"
                  placeholder="password"
                  required={true}
                  type="password"
                />
                <label
                  className="login-field-icon fui-lock"
                  htmlFor="login-pass"
                />
              </div>
              <button
                className="btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </form>
            <Link
              replace={false}
              to="/register"
            >
               Register new Account
            </Link>
          </div>,
        ],
        "className": "login-screen",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Login
            </h1>,
            "className": "app-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
            },
            "ref": null,
            "rendered": "Login",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <form
                className="register-form"
                method="POST"
                name="login_form"
                onSubmit={[Function]}
              >
                <div
                  className="control-group"
                >
                  <input
                    className="login-field"
                    name="email"
                    placeholder="email"
                    required={true}
                    type="text"
                  />
                  <label
                    className="login-field-icon fui-user"
                    htmlFor="login-name"
                  />
                </div>
                <div
                  className="control-group"
                >
                  <input
                    className="login-field"
                    name="password"
                    placeholder="password"
                    required={true}
                    type="password"
                  />
                  <label
                    className="login-field-icon fui-lock"
                    htmlFor="login-pass"
                  />
                </div>
                <button
                  className="btn btn-primary"
                  type="submit"
                >
                  Submit
                </button>
              </form>,
              <Link
                replace={false}
                to="/register"
              >
                 Register new Account
              </Link>,
            ],
            "className": "login-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="control-group"
                  >
                    <input
                      className="login-field"
                      name="email"
                      placeholder="email"
                      required={true}
                      type="text"
                    />
                    <label
                      className="login-field-icon fui-user"
                      htmlFor="login-name"
                    />
                  </div>,
                  <div
                    className="control-group"
                  >
                    <input
                      className="login-field"
                      name="password"
                      placeholder="password"
                      required={true}
                      type="password"
                    />
                    <label
                      className="login-field-icon fui-lock"
                      htmlFor="login-pass"
                    />
                  </div>,
                  <button
                    className="btn btn-primary"
                    type="submit"
                  >
                    Submit
                  </button>,
                ],
                "className": "register-form",
                "method": "POST",
                "name": "login_form",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="login-field"
                        name="email"
                        placeholder="email"
                        required={true}
                        type="text"
                      />,
                      <label
                        className="login-field-icon fui-user"
                        htmlFor="login-name"
                      />,
                    ],
                    "className": "control-group",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "login-field",
                        "name": "email",
                        "placeholder": "email",
                        "required": true,
                        "type": "text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "login-field-icon fui-user",
                        "htmlFor": "login-name",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "label",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="login-field"
                        name="password"
                        placeholder="password"
                        required={true}
                        type="password"
                      />,
                      <label
                        className="login-field-icon fui-lock"
                        htmlFor="login-pass"
                      />,
                    ],
                    "className": "control-group",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "login-field",
                        "name": "password",
                        "placeholder": "password",
                        "required": true,
                        "type": "password",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "login-field-icon fui-lock",
                        "htmlFor": "login-pass",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "label",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Submit",
                    "className": "btn btn-primary",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": "button",
                },
              ],
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " Register new Account",
                "replace": false,
                "to": "/register",
              },
              "ref": null,
              "rendered": " Register new Account",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="login-screen"
        >
          <div
            className="app-title"
          >
            <h1>
              Login
            </h1>
          </div>
          <div
            className="login-form"
          >
            <form
              className="register-form"
              method="POST"
              name="login_form"
              onSubmit={[Function]}
            >
              <div
                className="control-group"
              >
                <input
                  className="login-field"
                  name="email"
                  placeholder="email"
                  required={true}
                  type="text"
                />
                <label
                  className="login-field-icon fui-user"
                  htmlFor="login-name"
                />
              </div>
              <div
                className="control-group"
              >
                <input
                  className="login-field"
                  name="password"
                  placeholder="password"
                  required={true}
                  type="password"
                />
                <label
                  className="login-field-icon fui-lock"
                  htmlFor="login-pass"
                />
              </div>
              <button
                className="btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </form>
            <Link
              replace={false}
              to="/register"
            >
               Register new Account
            </Link>
          </div>
        </div>,
        "className": "login",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="app-title"
            >
              <h1>
                Login
              </h1>
            </div>,
            <div
              className="login-form"
            >
              <form
                className="register-form"
                method="POST"
                name="login_form"
                onSubmit={[Function]}
              >
                <div
                  className="control-group"
                >
                  <input
                    className="login-field"
                    name="email"
                    placeholder="email"
                    required={true}
                    type="text"
                  />
                  <label
                    className="login-field-icon fui-user"
                    htmlFor="login-name"
                  />
                </div>
                <div
                  className="control-group"
                >
                  <input
                    className="login-field"
                    name="password"
                    placeholder="password"
                    required={true}
                    type="password"
                  />
                  <label
                    className="login-field-icon fui-lock"
                    htmlFor="login-pass"
                  />
                </div>
                <button
                  className="btn btn-primary"
                  type="submit"
                >
                  Submit
                </button>
              </form>
              <Link
                replace={false}
                to="/register"
              >
                 Register new Account
              </Link>
            </div>,
          ],
          "className": "login-screen",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                Login
              </h1>,
              "className": "app-title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
              },
              "ref": null,
              "rendered": "Login",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <form
                  className="register-form"
                  method="POST"
                  name="login_form"
                  onSubmit={[Function]}
                >
                  <div
                    className="control-group"
                  >
                    <input
                      className="login-field"
                      name="email"
                      placeholder="email"
                      required={true}
                      type="text"
                    />
                    <label
                      className="login-field-icon fui-user"
                      htmlFor="login-name"
                    />
                  </div>
                  <div
                    className="control-group"
                  >
                    <input
                      className="login-field"
                      name="password"
                      placeholder="password"
                      required={true}
                      type="password"
                    />
                    <label
                      className="login-field-icon fui-lock"
                      htmlFor="login-pass"
                    />
                  </div>
                  <button
                    className="btn btn-primary"
                    type="submit"
                  >
                    Submit
                  </button>
                </form>,
                <Link
                  replace={false}
                  to="/register"
                >
                   Register new Account
                </Link>,
              ],
              "className": "login-form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="control-group"
                    >
                      <input
                        className="login-field"
                        name="email"
                        placeholder="email"
                        required={true}
                        type="text"
                      />
                      <label
                        className="login-field-icon fui-user"
                        htmlFor="login-name"
                      />
                    </div>,
                    <div
                      className="control-group"
                    >
                      <input
                        className="login-field"
                        name="password"
                        placeholder="password"
                        required={true}
                        type="password"
                      />
                      <label
                        className="login-field-icon fui-lock"
                        htmlFor="login-pass"
                      />
                    </div>,
                    <button
                      className="btn btn-primary"
                      type="submit"
                    >
                      Submit
                    </button>,
                  ],
                  "className": "register-form",
                  "method": "POST",
                  "name": "login_form",
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          className="login-field"
                          name="email"
                          placeholder="email"
                          required={true}
                          type="text"
                        />,
                        <label
                          className="login-field-icon fui-user"
                          htmlFor="login-name"
                        />,
                      ],
                      "className": "control-group",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "login-field",
                          "name": "email",
                          "placeholder": "email",
                          "required": true,
                          "type": "text",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "login-field-icon fui-user",
                          "htmlFor": "login-name",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "label",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          className="login-field"
                          name="password"
                          placeholder="password"
                          required={true}
                          type="password"
                        />,
                        <label
                          className="login-field-icon fui-lock"
                          htmlFor="login-pass"
                        />,
                      ],
                      "className": "control-group",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "login-field",
                          "name": "password",
                          "placeholder": "password",
                          "required": true,
                          "type": "password",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "login-field-icon fui-lock",
                          "htmlFor": "login-pass",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "label",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Submit",
                      "className": "btn btn-primary",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Submit",
                    "type": "button",
                  },
                ],
                "type": "form",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": " Register new Account",
                  "replace": false,
                  "to": "/register",
                },
                "ref": null,
                "rendered": " Register new Account",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
